SyslogNet.Client/Transport/SyslogLocalSender.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using SyslogNet.Client.Serialization;
+
+namespace SyslogNet.Client.Transport
+{
+	public class SyslogLocalSender : ISyslogMessageSender, IDisposable
+	{
+		// Will not work on Windows, as this relies on Unix system calls
+		public SyslogLocalSender()
+		{
+			PlatformID platform = Environment.OSVersion.Platform;
+			if (!(platform == PlatformID.MacOSX || platform == PlatformID.Unix)) {
+				throw new CommunicationsException("SyslogLocalSender is only available on Unix-like systems (e.g., Linux, BSD, OS X)");
+			}
+		}
+
+		[DllImport("libc", ExactSpelling=true)]

```

Comment:
Maybe we can use Mono calls here? https://github.com/mono/mono/blob/master/mcs/class/Mono.Posix/Mono.Unix.Native/Syscall.cs